File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int min = Integer.MAX_VALUE,curr,currMax=0,max=0;
        for(int price :prices){
            curr = price;
            min = Math.min(curr,min);
            max = Math.max(curr-min,max); 
        }
        return max;
        
        
        
    }
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        if len(prices)==1:
            return 0
        left = 0
        right = 0
        max_profit = 0
        while right<len(prices):
            # print(f"{left} {right}")
            profit = prices[right] - prices[left]
            if profit < 0:
                left += 1
                continue
            if profit>0:
                max_profit = max(max_profit,profit)
            right += 1
        return max_profit

